nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,848 @@
+                          'HERTZ',
+                          'METER',
+                          'RADIAN')
+
+CIFTI_BrainStructures = ('CIFTI_STRUCTURE_ACCUMBENS_LEFT',
+                         'CIFTI_STRUCTURE_ACCUMBENS_RIGHT',
+                         'CIFTI_STRUCTURE_ALL_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_ALL_GREY_MATTER',
+                         'CIFTI_STRUCTURE_AMYGDALA_LEFT',
+                         'CIFTI_STRUCTURE_AMYGDALA_RIGHT',
+                         'CIFTI_STRUCTURE_BRAIN_STEM',
+                         'CIFTI_STRUCTURE_CAUDATE_LEFT',
+                         'CIFTI_STRUCTURE_CAUDATE_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLUM',
+                         'CIFTI_STRUCTURE_CEREBELLUM_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLUM_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CORTEX',
+                         'CIFTI_STRUCTURE_CORTEX_LEFT',
+                         'CIFTI_STRUCTURE_CORTEX_RIGHT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_LEFT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_RIGHT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_LEFT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_RIGHT',
+                         'CIFTI_STRUCTURE_OTHER',
+                         'CIFTI_STRUCTURE_OTHER_GREY_MATTER',
+                         'CIFTI_STRUCTURE_OTHER_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_PALLIDUM_LEFT',
+                         'CIFTI_STRUCTURE_PALLIDUM_RIGHT',
+                         'CIFTI_STRUCTURE_PUTAMEN_LEFT',
+                         'CIFTI_STRUCTURE_PUTAMEN_RIGHT',
+                         'CIFTI_STRUCTURE_THALAMUS_LEFT',
+                         'CIFTI_STRUCTURE_THALAMUS_RIGHT')
+
+
+def _value_if_klass(val, klass, none_ok=True):
+    if none_ok and val is None:
+        return val
+    elif isinstance(val, klass):
+        return val
+    else:
+        raise ValueError('Not a valid %s instance.' % klass.__name__)
+
+
+def _value_or_make_klass(val, klass):
+    if val is None:
+        return klass()
+    else:
+        return _value_if_klass(val, klass)
+
+
+def _underscore(string):
+    """ Convert a string from camelcase to underscored """
+    string = re.sub(r'([A-Z]+)([A-Z][a-z])', r'\1_\2', string)
+    return re.sub(r'([a-z0-9])([A-Z])', r'\1_\2', string).lower()
+
+
+class Cifti2MetaData(xml.XmlSerializable):
+    """ A list of key-value pairs stored in the list self.data """
+
+    def __init__(self, nvpair=None):
+        self.data = []
+        self.add_metadata(nvpair)
+
+    def _add_remove_metadata(self, metadata, func):
+        pairs = []
```

Comment:
A little ugly having one function for both add and remove.

Why not refactor into function `def _check_nvpairs(self, metadata)` then:

```
def add_metadata(self, nvpair):
    for pair in self._check_nvpairs(nvpair):
        self.data.append(pair)
```